File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    sumHash = 0
    hashDict = {}
    for part in participant:
        hashDict[hash(part)]=part
        sumHash+=hash(part)

    for comp in completion:
        sumHash-=hash(comp)
    
    answer = hashDict[sumHash]
    return answer

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;
import java.io.*;

class Solution {
    public String solution(String[] participant, String[] completion) {            
        String answer = "";
        Map<String,Integer> map = new HashMap<>();
        for(int i=0;i<participant.length;i++){
            if(map.containsKey(participant[i])){
                int a=map.remove(participant[i]);
                map.put(participant[i],a+1);
            }else{map.put(participant[i],1);}
        }
        for(int i=0;i<completion.length;i++){
            int b= map.remove(completion[i])-1;
            if(b>0){map.put(completion[i],b);}
        }
        for(String next:map.keySet()){
            answer+=next;
        }
        return answer;
    }
}